modules/gateway/nsg.tf

```diff
@@ -0,0 +1,100 @@
+  resource_group_name         = var.resource_group_name
+  network_security_group_name = azurerm_network_security_group.graphdb_gateway.name
+
+  name                         = "AllowLoadBalancer"
+  description                  = "Allows AzureLoadBalancer to perform balancing to gateway instances"
+  priority                     = 110
+  direction                    = "Inbound"
+  access                       = "Allow"
+  protocol                     = "Tcp"
+  source_address_prefix        = "AzureLoadBalancer"
+  source_port_range            = "*"
+  destination_address_prefixes = var.gateway_subnet_address_prefixes
+  destination_port_range       = "*"
+}
+
+resource "azurerm_network_security_rule" "graphdb_gateway_allow_inbound_address" {
+  count = length(var.gateway_allowed_address_prefixes) == 0 ? 1 : 0
+
+  resource_group_name         = var.resource_group_name
+  network_security_group_name = azurerm_network_security_group.graphdb_gateway.name
+
+  name                         = "AllowInBoundAddress"
+  priority                     = 200
+  direction                    = "Inbound"
+  access                       = "Allow"
+  protocol                     = "Tcp"
+  source_address_prefix        = var.gateway_allowed_address_prefix
+  source_port_range            = "*"
+  destination_address_prefixes = var.gateway_subnet_address_prefixes
+  destination_port_ranges      = [80, 443]
+}
+
+resource "azurerm_network_security_rule" "graphdb_gateway_allow_inbound_addresses" {
+  count = length(var.gateway_allowed_address_prefixes) > 0 ? 1 : 0
+
+  resource_group_name         = var.resource_group_name
+  network_security_group_name = azurerm_network_security_group.graphdb_gateway.name
+
+  name                         = "AllowInBoundAddresses"
+  priority                     = 201
+  direction                    = "Inbound"
+  access                       = "Allow"
+  protocol                     = "Tcp"
+  source_address_prefixes      = var.gateway_allowed_address_prefixes
+  source_port_range            = "*"
+  destination_address_prefixes = var.gateway_subnet_address_prefixes
+  destination_port_ranges      = [80, 443]
+}
+
+resource "azurerm_network_security_rule" "graphdb_gateway_deny_inbound" {
+  resource_group_name         = var.resource_group_name
+  network_security_group_name = azurerm_network_security_group.graphdb_gateway.name
+
+  name                         = "DenyInBound"
+  description                  = "Denies any other inbound traffic to the associated subnet"
+  priority                     = 4096
+  direction                    = "Inbound"
+  access                       = "Deny"
+  protocol                     = "*"
+  source_address_prefix        = "*"
+  source_port_range            = "*"
+  destination_address_prefixes = var.gateway_subnet_address_prefixes
+  destination_port_range       = "*"
+
+  # Otherwise Azure will reject the deny rule
+  depends_on = [
```

Comment:
Even with the `depends_on` Azure would complain that `65200-65535` are blocked meaning the creation is async behind the API..